components/raftstore/src/store/fsm/peer.rs

```diff
@@ -2219,6 +2245,28 @@ where
             let campaigned = new_peer.peer.maybe_campaign(is_leader);
             new_peer.has_ready |= campaigned;
 
+            // We roughly estimate the size and keys for new region to avoid blocking heartbeat.
+            // But the value may be far from the real value. So we let split checker to update
+            // it immediately.
+            if !self.fsm.peer.is_region_size_or_keys_none() {
+                new_peer.peer.approximate_size =
+                    Some(self.fsm.peer.approximate_size.unwrap() / new_region_count);
+                new_peer.peer.approximate_keys =
+                    Some(self.fsm.peer.approximate_keys.unwrap() / new_region_count);
```

Comment:
you have refresh it. So do not need to divide again.

Reply:
Oh no，that's really a bug. I have fixed it.